aten/src/ATen/core/ivalue_inl.h

```diff
@@ -318,16 +318,8 @@ struct C10_EXPORT ivalue::Future final : c10::intrusive_ptr_target {
  public:
   explicit Future(TypePtr type, std::vector<c10::Device> devices={})
       : type_(std::move(type)),
-        impl_(getImplForDevices(devices)),
-        devices_(getSortedIndicesOfDevices(impl_, devices)) {
-    // Use current device to initialize currentDevice_. This is necessary
-    // because preMarkCompletedHook won't be called when the Future contains
-    // an error. Uninitialized currentDevice_ could lead to crash when used
-    // in CUDAGuard.
-    if (impl_ != nullptr) {
-      currentDevice_ = impl_->getDevice().index();
```

Comment:
According to the comment the reason for this was to avoid setting the current device to a "wrong" device. After this change, since we use an OptionalDeviceGuard, in such a case we would just not set the current device at all. This does indeed mean that the current device remains set to whatever it was before, hence it is indeed unspecified. Do you think this will be an issue?